pkg-manager/hoist/src/index.ts

```diff
@@ -117,6 +86,33 @@ export function getHoistedDependencies (opts: GetHoistedDependenciesOpts) {
 
   const getAliasHoistType = createGetAliasHoistType(opts.publicHoistPattern, opts.privateHoistPattern)
 
+  if (opts.hoistWorkspaceProjects) {
+    const allProjects = Object.keys(opts.lockfile.importers)
+
+    const workspaceDir = path.dirname(opts.publicHoistedModulesDir)
+
+    const hoistedWorkspaceProjects =
+        allProjects
+          // Ignore the "." project because it is the root project
+          .filter((projectPath) => projectPath !== '.')
+          .map((projectPath) => {
+            // eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
+            const manifest = require(path.join(workspaceDir, projectPath))
+            return {
+              projectPath,
+              hoistedProjectPath: path.join(opts.privateHoistedModulesDir, 'node_modules', manifest.name),
+            }
+          })
+
+    hoistedWorkspaceProjects.map((hoistedWorkspaceProject) => {
+      const exists = fs.existsSync(hoistedWorkspaceProject.hoistedProjectPath)
+      if (!exists) {
+        return symlinkdirSync(hoistedWorkspaceProject.projectPath, hoistedWorkspaceProject.hoistedProjectPath, { overwrite: false })
+      }
+      return true
+    })
+  }
+
   return hoistGraph(deps, opts.lockfile.importers['.']?.specifiers ?? {}, {
```

Comment:
hoistGraph should return the hoisted workspace projects too. The already existing symlinking logic should handle symlinking of this.

Reply:
Ok, so the changes I should make should just get the projects, right?

Reply:
maybe. I don't know if it will be enough to make it work. But that should be the solution.

Reply:
I added a commit to apply this, is not ready yet but I'll test it tomorrow morning.